app/code/Magento/Multishipping/Test/Mftf/Test/StorefrontCreateOrderWithMultishippingAfterReturningToCartTest.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontCreateOrderWithMultishippingAfterReturningToCartTest">
+        <annotations>
+            <features value="Multishipping"/>
+            <stories value="Checkout with multiple addresses."/>
+            <title value="Checkout with multiple addresses after returning on cart page during checkout."/>
+            <description value="Verify customer able to checkout with multiple addresses after returning to cart page and continue checkout with browser 'back' button."/>
+            <severity value="AVERAGE"/>
+            <testCaseId value="MC-39583"/>
+            <useCaseId value="MC-36425"/>
+            <group value="multishipping"/>
+        </annotations>
+        <before>
+            <!--Create test data.-->
+            <createData entity="SimpleProduct2" stepKey="product"/>
+            <createData entity="Simple_US_Customer_Two_Addresses" stepKey="customer"/>
+            <!--Set up configuration.-->
+            <magentoCLI command="config:set {{EnableFreeShippingMethod.path}} {{EnableFreeShippingMethod.value}}" stepKey="enableFreeShipping"/>
+            <magentoCLI command="config:set {{EnableFlatRateShippingMethod.path}} {{EnableFlatRateShippingMethod.value}}" stepKey="enableFlatRateShipping"/>
+            <magentoCLI command="config:set {{EnableCheckMoneyOrderPaymentMethod.path}} {{EnableCheckMoneyOrderPaymentMethod.value}}" stepKey="enableCheckMoneyOrderPaymentMethod"/>
+        </before>
+        <after>
+            <!--Clean up test data, revert configuration.-->
+            <deleteData createDataKey="product" stepKey="deleteProduct"/>
+            <deleteData createDataKey="customer" stepKey="deleteCustomer"/>
+            <magentoCLI command="config:set {{DisableFreeShippingMethod.path}} {{DisableFreeShippingMethod.value}}" stepKey="disableFreeShipping"/>
```

Comment:
That's so awesome that we don't clean cache after setting a config via CLI anymore (it's invalidated automatically in this case). 

Great job!